core/iwasm/compilation/aot_llvm_extra.cpp

```diff
@@ -354,7 +354,9 @@ aot_apply_llvm_new_pass_manager(AOTCompContext *comp_ctx, LLVMModuleRef module)
                 MPM.addPass(PB.buildLTODefaultPipeline(OL, NULL));
         }
         else {
-            MPM.addPass(PB.buildPerModuleDefaultPipeline(OL));
+            MPM.addPass(OL == OptimizationLevel::O0
+                            ? PB.buildO0DefaultPipeline(OL)
+                            : PB.buildPerModuleDefaultPipeline(OL));
```

Comment:
The above buildLTOPreLinkDefaultPipeline/buildLTODefaultPipeline also cannot be used for O0, could we also fix it, seems it can be:
```C
            if (OL == OptimizationLevel::O0)
                MPM.addPass(PB.buildO0DefaultPipeline(OL, true));
            else if (comp_ctx->comp_data->func_count >= 10
                     || comp_ctx->enable_llvm_pgo || comp_ctx->use_prof_file)
                ..
            else
```